From a660df8338b5faf3a9f21bc7bdbd0a3ceb3c71f5 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 28 Aug 2013 14:07:59 +0800
Subject: [PATCH 36/43] ENGR00276301 USB: fix usb tethering over 3G mobile
 issue

refactor requestOrSendDataCallList function to fill linkaddress,dns,
gateway in response message.This will be got by framework to build
ppp0's LinkProperties.If missing this info, tethering will not get
the right operating interface according to its LinkProperties's route.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 reference-ril/reference-ril.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/reference-ril/reference-ril.c b/reference-ril/reference-ril.c
index e8fef81..2a0a9f4 100755
--- a/reference-ril/reference-ril.c
+++ b/reference-ril/reference-ril.c
@@ -804,13 +804,15 @@ static void requestOrSendDataCallList(RIL_Token *t)
                 char  propName[PROP_NAME_MAX];
                 memset(dnslist, 0, 128);
                 property_get("net.ppp0.dns1", propName, "8.8.8.8");
-                strlcat(dnslist, propValue, dnslist_sz);
+                strlcat(dnslist, propName, dnslist_sz);
                 strlcat(dnslist, separator, dnslist_sz);
                 property_get("net.ppp0.dns2", propName, "8.8.4.4");
-                strlcat(dnslist, propValue, dnslist_sz);
-
+                strlcat(dnslist, propName, dnslist_sz);
                 responses[i].dnses = dnslist;
-                responses[i].gateways = "";
+                property_get("net.ppp0.remote-ip", propName, "");
+                responses[i].gateways = propName;
+                property_get("net.ppp0.local-ip", propName, "");
+                responses[i].addresses = propName;
             }
         }
     }
-- 
1.8.0

